tools/gmhazard_scripts/db_creation/empirical_db/common.py

```diff
@@ -220,16 +226,39 @@ def __process_rupture(
                 str(im_type),
                 psa_periods if im_type is IMType.pSA else None,
             )
-            if not im_type is IMType.pSA:
+            if im_type is not IMType.pSA:
                 values = [values]
+            elif use_directivity:
+                nhm_fault = nhm_dict[rupture.rupture_name]
+                planes, lon_lat_depth = gc_rupture.get_fault_header_points(nhm_fault)
+                site_coords = np.asarray([stat_df.loc[station_name].values])
+                (
+                    fdi,
+                    _,
+                    phi_red,
+                ) = sha_calc.directivity.bea20.directivity.compute_fault_directivity(
+                    lon_lat_depth,
+                    planes,
+                    site_coords,
+                    25,
+                    4,
+                    nhm_fault.mw,
+                    nhm_fault.rake,
+                    periods=psa_periods,
+                )
             for i, value in enumerate(values):
                 full_im_name = (
                     IM(im_type, period=psa_periods[i])
                     if im_type is IMType.pSA
                     else IM(im_type)
                 )
-                mean = np.log(value[0])
-                stdev, sigma_inter, sigma_intra = value[1]
+                if use_directivity and im_type is IMType.pSA:
```

Comment:
I'd do the mean / sigma calculation first and then the directivity adjustment. Just so each block of code isn't doing too many steps at once.